static/js/common/fxa-login.js

```diff
@@ -5,6 +5,26 @@
         oauthHost: config.oauthHost,
     });
 
+    function currentPath() {
+        return location.pathname + location.search + location.hash;
+    }
+
+    function urlsafe(str) {
+        // This function makes a base64 string URL safe using python's base64
+        // module's replacements.
+        // https://docs.python.org/2/library/base64.html#base64.urlsafe_b64encode
```

Comment:
I think python's `urlsafe_b64encode` adds equal signs (which seems wrong). I guess that doesn't apply anywhere in your patch? Typically what you need to do is strip the equal signs to make it URL safe.